cpp/src/arrow/python/flight.h

```diff
@@ -35,14 +35,22 @@ namespace flight {
 /// Python.
 class ARROW_PYTHON_EXPORT PyFlightServerVtable {
  public:
-  std::function<void(PyObject*, const arrow::flight::FlightDescriptor&,
-                     std::unique_ptr<arrow::flight::FlightInfo>*)>
+  // Must return int so Cython will propagate exceptions
```

Comment:
I don't think you need this, you can just annotate the Cython function with `except *`. See https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#error-return-values

Reply:
Good catch, thanks.